### PR TITLE
Add OCP 4.2 scale run environment details

### LINKED ISSUE #20868: Tested Maximum page should be more specific
### Which section(s) is the issue in?
https://github.com/openshift/openshift-docs/blob/master/modules/openshift-cluster-maximums.adoc

### What needs fixing?
openshift-cluster-maximums.adoc Should become `openshift-<platform>-cluster-maximums.adoc` and contain how did we achieve a specific scale for that specific platform. 

Today our documentation only discusses what we achieved, how what we achieved things on.

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit:
- Adds details about the environment including cloud platform used
  for running the performance and scale tests on OCP 4.2 as well as
  info about the cluster configuration like instance types, CPU,
  memory, disk, node count e.t.c.
- Adds info on why we went with particular configuration as well as
  pointers to the docs wherever required.

Fixes #20868

### CODE REVIEW

**Comment by chaitanyaenr** [comment]
@ahardin-rh @jtaleric @mffiedler PTAL.

**Comment by jtaleric** [comment]
@chaitanyaenr add to the commit message : `fixes #20868`

File: modules/openshift-cluster-maximums-environment.adoc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+| Master/Etcd
+| n1-highmem-16
+| 16
+| 104
+| Regional/Zonal SSD 
+| 220
+| 3
+| us-east4
+
+| Infra footnoteref:[infranodesgcp,Infra nodes are used to host openshift-monitoring, ingress and router components to make sure they have enough resources to run at large scale. Procedure on creating Infra machinesets can be found at https://docs.openshift.com/container-platform/4.2/machine_management/creating-infrastructure-machinesets.html.]
```
Comment (ahardin-rh) [inline modules/openshift-cluster-maximums-environment.adoc:23]: Infra nodes are used to host openshift-monitoring, Ingress, and router components to ensure they have enough resources to run at a large scale.

Unfortunately, we have to remove the link to the procedure, as we are currently not allowed to include internal xrefs within modules. Perhaps we can move this to a new "Related Information" section of the "assembly" file (as seen in [this template](Related Information)), which in this case is the planning-your-environment-according-to-object-maximums.adoc.

File: modules/openshift-cluster-maximums-environment.adoc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+| n1-standard-64
+| 64
+| 240
+| Regional/Zonal SSD 
+| 100 
+| 3
+| us-east4
+
+| Workload footnoteref:[workloadnodegcp, Workload node is dedicated to run performance/scalability workload generators.]
```
Comment (ahardin-rh) [inline modules/openshift-cluster-maximums-environment.adoc:32]: performance and scalability

File: modules/openshift-cluster-maximums-environment.adoc [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+| n1-standard-16 
+| 16
+| 60
+| Regional/Zonal SSD
+| 500 footnoteref:[disksizegcp, Larger disk size is used to have enough space to store large amounts of data collected during the performamce/scalability test run.]
```
Comment (ahardin-rh) [inline modules/openshift-cluster-maximums-environment.adoc:37]: performance and scalability 
(typo in "performance" as well"

**Comment by ahardin-rh** [comment]
@chaitanyaenr Thanks for the edits! 
@mffiedler Can you please provide a QE review?

**Review by mffiedler — APPROVED** [review]
/lgtm

**Comment by ahardin-rh** [comment]
Thank you!

**Pull request merged**